include/hipSYCL/sycl/libkernel/id.hpp

```diff
@@ -283,6 +283,45 @@ struct id {
   HIPSYCL_ID_BINARY_OP_SIZE_T(&)
   HIPSYCL_ID_BINARY_OP_SIZE_T(|)
   HIPSYCL_ID_BINARY_OP_SIZE_T(^)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(&&)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(||)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(<)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(>)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(<=)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(>=)
+
+  // Unary operators +,-
+  friend id operator+(const id& rhs) {
+    return rhs;
+  }
+
+  friend id operator-(const id& rhs) {
+    return -1*rhs;
```

Comment:
Does that make sense? `id` stores `size_t`, so an unsigned type.

Reply:
I just added what's in the spec, but you're totally right, this makes no sense...
I will open an issue in the SYCL-Docs repo and ask about this

Reply:
Hmm, reading up on this a bit, I think this might make sense or is at least well-defined. The C++ standard says
> The negative of an unsigned quantity is computed by subtracting its value from $2^n$, where n is the number of bits in the promoted operand.

So I guess we can just keep it to be conformant